### PR TITLE
Deck import: Pre-validate zip file

### LINKED ISSUE #6489: Don't show an exception report dialog when provided with known invalid zip files
Cause: Investigations into #6419

Assuming these do not imply a problem with the library, then these issues cause noise in ACRA, and will lead to a degraded user experience. 

But as we now have the report data, we can classify the exceptions based on the exception details to provide better information to help the user.

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/ab7bfc68-30d0-4129-a4d2-04427f4d3123

```
java.util.zip.ZipException: archive is not a ZIP archive
at org.apache.commons.compress.archivers.zip.ZipFile.positionAtEndOfCentralDirectoryRecord(ZipFile.java:2)
at org.apache.commons.compress.archivers.zip.ZipFile.positionAtCentralDirectory(ZipFile.java:1)
at org.apache.commons.compress.archivers.zip.ZipFile.populateFromCentralDirectory(ZipFile.java:2)
at org.apache.commons.compress.archivers.zip.ZipFile.<init>(ZipFile.java:19)
at org.apache.commons.compress.archivers.zip.ZipFile.<init>(ZipFile.java:4)
at org.apache.commons.compress.archivers.zip.ZipFile.<init>(ZipFile.java:1)
at com.ichi2.async.CollectionTask.doInBackgroundImportReplace(CollectionTask.java:9)
at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:37)
at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:1)
at android.os.AsyncTask$3.call(AsyncTask.java:378)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:919)
```

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
Previously, we performed the check inside either `AnkiPackageImporter`/`doInBackgroundReplace`. This caused exception reports to be thrown when zip files were invalid, which wasn't a great UX

## Fixes
Fixes #6489

## Approach
We move to `ImportUtil` to validate the import, which displays an error and does not raise an exception report.

## How Has This Been Tested?

On my phone:

* Empty file - no exception report, "Failed to unzip apkg: archive is not a ZIP archive"
* Corrupt file - no exception report, "Failed to unzip apkg: central directory is empty ..."
* Good file - imports as normal

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.java [hunk 1]
```diff
@@ -173,12 +174,32 @@ private String handleContentProviderFile(Context context, Intent intent, Uri dat
...
             }
         }
 
 
+        protected String validateZipFile(Context context, String filePath) {
+            File file = new File(filePath);
+            try {
+                new ZipFile(file);
+            } catch (Exception e) {
+                Timber.e(e, "Failed to validate zip");
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.java:196]: Pretty much a nit but 'e' is for "application has unknown state and we're basically a contained explosion at this point", 'w' is for, input was unexpected and it definitely messes with what the user was trying to do, but internal state is fine, we know what's going on, we're good'

This is 'w' I think

**Comment by mikehardy** [comment]
release engineering note: I would pick this if we do anything else on release-2.11 otherwise it feeds 2.12 of course

**Comment by david-allison-1** [comment]
Currently, the "Import" menu item means nothing to new users. 

There are users trying to import `.docx` files, which will still get through quite a lot of the validation, as they're valid zips. This will probably require reorganisation of the deck picker to solve.  That'd be my next goal in the area

**Comment by mikehardy** [comment]
1- import docx wow, would have never thought of that, but of course? Like I have a spreadsheet with 2 columns, why can't I import?
2- reorganize deck picker where's the giphy for there be dragons, but yeah, picking that beast apart would be fantastic

**Pull request merged**